src/index.js

```diff
@@ -88,9 +88,18 @@ const ChartComponent = React.createClass({
 		nextData.forEach(function (dataset, sid) {
 			if (currentData[sid] && currentData[sid].data) {
 				currentData[sid].data.splice(nextData[sid].data.length);
+
 				dataset.data.forEach(function (point, pid) {
 					currentData[sid].data[pid] = nextData[sid].data[pid];
 				});
+
+				delete dataset.data;
```

Comment:
Technically... yes.

But since we're building a new version of the dataset on every render for now this is sadly the only way of making sure we don't miss new properties that get passed in without doing a bunch of deepEquals. In the future I would like to make this more immutable, but for now this was the fastest way to get a fix out the door. :)